pkg/genericapiserver/server/storage_factory.go

```diff
@@ -74,7 +74,19 @@ type DefaultStorageFactory struct {
 	APIResourceConfigSource APIResourceConfigSource
 
 	// newStorageCodecFn exists to be overwritten for unit testing.
-	newStorageCodecFn func(storageMediaType string, ns runtime.StorageSerializer, storageVersion, memoryVersion schema.GroupVersion, config storagebackend.Config) (codec runtime.Codec, err error)
+	newStorageCodecFn func(opts StorageCodecOptions) (codec runtime.Codec, err error)
+}
+
+// StorageCodecOptions are the arguments passed to newStorageCodecFn
+type StorageCodecOptions struct {
```

Comment:
This should be called `StorageCodecConfig`, compare the other structs in `config.go`.